api/src/shared/infrastructure/repositories/user-repository.js

```diff
@@ -295,32 +280,29 @@ const updateHasSeenLevelSevenInfoToTrue = async function (id) {
 };
 
 const updateHasSeenNewDashboardInfoToTrue = async function (id) {
-  const user = await BookshelfUser.where({ id }).fetch({ require: false });
-  await user.save({ hasSeenNewDashboardInfo: true }, { patch: true, method: 'update' });
-  return bookshelfToDomainConverter.buildDomainObject(BookshelfUser, user);
+  const [user] = await knex('users').where({ id }).update('hasSeenNewDashboardInfo', true).returning('*');
+
+  return new User(user);
 };
 
 const updateHasSeenChallengeTooltip = async function ({ userId, challengeType }) {
-  const user = await BookshelfUser.where({ id: userId }).fetch({ require: false });
+  let user;
   if (challengeType === 'focused') {
-    await user.save({ hasSeenFocusedChallengeTooltip: true }, { patch: true, method: 'update' });
+    [user] = await knex('users').where({ id: userId }).update('hasSeenFocusedChallengeTooltip', true).returning('*');
   }
   if (challengeType === 'other') {
-    await user.save({ hasSeenOtherChallengesTooltip: true }, { patch: true, method: 'update' });
+    [user] = await knex('users').where({ id: userId }).update('hasSeenOtherChallengesTooltip', true).returning('*');
   }
-  return bookshelfToDomainConverter.buildDomainObject(BookshelfUser, user);
+  return new User(user);
 };
 
 const acceptPixLastTermsOfService = async function (id) {
-  const user = await BookshelfUser.where({ id }).fetch({ require: false });
-  await user.save(
-    {
-      lastTermsOfServiceValidatedAt: new Date(),
-      mustValidateTermsOfService: false,
-    },
-    { patch: true, method: 'update' },
-  );
-  return bookshelfToDomainConverter.buildDomainObject(BookshelfUser, user);
+  const [user] = await knex('users')
+    .where({ id })
+    .update({ lastTermsOfServiceValidatedAt: new Date(), mustValidateTermsOfService: false })
+    .returning('*');
```

Comment:
Bookshelf pouvait se charger de modifier automatiquement le `updatedAt` d'un enregistrement, mais avec Knex c'est à faire manuellement à chaque requête effectuant une modification :

```suggestion
  const [user] = await knex('users')
    .where({ id })
    .update({
      lastTermsOfServiceValidatedAt: new Date(),
      mustValidateTermsOfService: false,
      updatedAt: new Date(),
    })
    .returning('*');
```